### PR TITLE
Conversion nudging params parsing

### PR DESCRIPTION
**Pull request by leifdenby**
This pull-request is meant to replace work in https://github.com/EUREC4A-UK/lagtraj/pull/191

### CODE REVIEW

File: tests/test_forcing_nudging.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+VALID_YAML_EXAMPLES = [
+    """
+nudging_method_scalars: constant
+nudging_timescale_scalars: 10
+""",
+    """
+nudging_method_scalars: runtime_inversion_height
+nudging_timescale_scalars: 10
+nudging_layer_thickness_scalars: 3000.0
```
Comment (sjboeing) [inline tests/test_forcing_nudging.py:52]: These values are valid, but maybe it is better to use something like 10800 sec for the timestep and 500m for thickness.

Comment (sjboeing) [reply tests/test_forcing_nudging.py:52]: What is unclear here is whether you can only provide scalars, or need momentum as well?

File: lagtraj/forcings/conversion/input_definitions_nudging.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def _construct_nudging_required_fields():
+    """
+    parameters for nudging scalars and momentum are set separately, so we add
+    them here in turn to the set of required parameters
+    """
+    fields = {}
+    for nudged_variable_groups in ["scalars", "momentum"]:
+        fields[
```
Comment (sjboeing) [inline lagtraj/forcings/conversion/input_definitions_nudging.py:29]: Is there a more informative variable name here?

**Review by sjboeing — COMMENTED** [review]
Looks exciting Leif! I haven't looked at the testing in detail, but happy to have another look later.

File: lagtraj/forcings/conversion/input_definitions_nudging.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+For nudging parameters we have quite a few sets of parameters and they can be
+used to set the parameters for nudging momentum and scalar forcing profiles
+separately (for example `nudging_timescale_scalars` and
+`nudging_timescale_momentum`). To avoid typos while enumerating all those
+options (and the cases for each nudging method were each parameter is required)
+this code takes care of generating the valid combinations.
+"""
+
+
+AVAILABLE_NUDGING_METHODS = [
```
Comment (sjboeing) [inline lagtraj/forcings/conversion/input_definitions_nudging.py:11]: Should we add "off", i.e. an explicit acknowledgement that nudging is turned off?

File: tests/test_forcing_nudging.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+nudging_shape_scalars: cos
+""",
+    """
+nudging_method_scalars: fixed_height
+nudging_timescale_scalars: 10
+nudging_layer_thickness_scalars: 3000.0
```
Comment (sjboeing) [inline tests/test_forcing_nudging.py:58]: I suggest changing "layer_thickness" to "transition_thickness", so it is explicit. Not sure if "height_above"/"lowest_height" is better for height.

File: lagtraj/forcings/conversion/input_definitions_nudging.py (hunk 2 above)
Comment (leifdenby) [reply lagtraj/forcings/conversion/input_definitions_nudging.py:29]: Rename to `nudging_parameters` thank you!

File: lagtraj/forcings/conversion/input_definitions_nudging.py (hunk 3 above)
Comment (leifdenby) [reply lagtraj/forcings/conversion/input_definitions_nudging.py:11]: Good idea

**Comment by sjboeing** [comment]
Superseded by #200

**Pull request closed without merging**